firebase-firestore/src/main/java/com/google/firebase/firestore/remote/FirestoreChannel.java

```diff
@@ -53,6 +55,11 @@
   private static final String X_GOOG_API_CLIENT_VALUE =
       "gl-java/ fire/" + BuildConfig.VERSION_NAME + " grpc/";
 
+  // This timeout is used when attempting to establish a connection. If a connection attempt
+  // does not succeed in time, we close the stream and restart the connection, rather than having
```

Comment:
The timer callback does not seem to "close the stream" in any way that I can tell. Based on comments in markChannelIdle, the first stream continues? Do this actually close the stream, or is this a side effect?

From what I can tell, the connectivityAttemptTimer merely starts a new clientCall (through a ~recursive invocation of runBidiStreamingRpc). What prevents the observer from getting callbacks from the old stream once that fails ~2 minutes later?

Reply:
I think you're looking at an old version of the PR? 😬   This code is gone in the latest version, right?

Reply:
Yes, that's the outdated version. The new one can be found [here](https://github.com/firebase/firebase-android-sdk/pull/1045/files#diff-03f10e7f8aea06522d1afa268c07d2cfR58).